### DISCUSSION TITLE
Generation of MR once a day

### DISCUSSION THREAD

**Discussion by lmilbaum** [renovatebot/renovate#29698] [Request Help]
### What would you like help with?

Other

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

gitlab

### Please tell us more about your question or problem

I'd like to force MR created once a day. It is needed for invoking a post upgrade command. Not sure if the command is invoked  if there is no dependency to update.
One option I'm considering is tracking a file which contains a date string.  Renovate can be used for bumping the date which will result in invoking the post upgrade command.
Is there a way to bump a date string?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by rarkins** [comment]
Try Custom manager, custom datasource, loose versioning (eg date is 20240615)

**Reply by lmilbaum** [reply]
Thanks @rarkins. Are you aware of a relevant datasource I could use?

**Reply by rarkins** [reply]
No

**Reply by lmilbaum** [reply]
OK. Found the following: `http://worldtimeapi.org/api/ip`.
It returns: 
```
curl http://worldtimeapi.org/api/ip         
{
  "abbreviation": "IDT",
  "client_ip": "xx.xx.xx.xx",
  "datetime": "2024-06-16T14:28:05.619908+03:00",
  "day_of_week": 0,
  "day_of_year": 168,
  "dst": true,
  "dst_from": "2024-03-29T00:00:00+00:00",
  "dst_offset": 3600,
  "dst_until": "2024-10-26T23:00:00+00:00",
  "raw_offset": 7200,
  "timezone": "Asia/Jerusalem",
  "unixtime": 1718537285,
  "utc_datetime": "2024-06-16T11:28:05.619908+00:00",
  "utc_offset": "+03:00",
  "week_number": 24
}
```

I assume the config should look something like this:
```
  "customDatasources": {
    "dates": {
      "defaultRegistryUrlTemplate": "http://worldtimeapi.org/api/ip",
      "transformTemplates": [
        "{\"releases\":\"\"}"
      ],
      "format": "json"
    }
  },
```

Can you please advice how to set the `transformTemplatees` key?

**Reply by lmilbaum** [reply]
The following repo is for testing it - https://github.com/platform-engineering-org-test/renovate-test.
The configuration is not working yet.

**Reply by lmilbaum** [reply]
@rarkins Would you be able to take a look?

**Reply by rarkins** [reply]
Sorry, I don't have time at the moment for anything in-depth unless it relates to a customer escalation

**Comment by francisbolduc** [comment]
I've stumbled on a similar need but wanted to update every year.
I've achieved this using the JSONata date functions `fromMillis` and `millis` (doc: <https://docs.jsonata.org/date-time>)
Also, I'm also using `"defaultRegistryUrlTemplate": "file://"` to prevent this "error" from renovate when used in a custom manager : `Failed to look up custom.year package <depName>`

every year:

```json
  "customDatasources": {
    "year": {
      "defaultRegistryUrlTemplate": "file://",
      "transformTemplates": [
        "{\"releases\": [{\"version\": $fromMillis($millis(), \"[Y0001]\")}]}"
      ]
    }
  }
```

every day (change format to your liking):

```json
  "customDatasources": {
    "day": {
      "defaultRegistryUrlTemplate": "file://",
      "transformTemplates": [
        "{\"releases\": [{\"version\": $fromMillis($millis(), \"[Y0001][M01][D01]\")}]}"
      ]
    }
  }
```

**Reply by lmilbaum** [reply]
nice.. thanks for sharing. I've switched to another service which returns the data information. At the moment it works nicely for us.
